BREAKING CHANGE: Updated to latest version of go-i18n to fix security vulnerability

Aerex opened the pull request:
# Context

Older versions of go-i18n has been flagged with a security vulnerability so the cli-sdk must be updated to at least v2.0.0 to resolve the vulnerability. There is an option to find another open source project however, go-i18n is the most supported and widely used localization package for Go.


# Breaking Changes
The API between v1.0 and v2.0 for go-i18n is drastically different. There are two points to keep in mind.


- The filename have been updated to match what go-i18n expects when parsing the generating bindata produced by go-bindata. Instead of `en_US.all.json` it has been changed to `all.en_US.json`. Any package that pulls the latest version of the CLI SDK will need to update the translation filenames accordingly. See files in i18n/resources/ for examples.

# Callouts

- The signature method remains the same as shown in the example below.
 ```
T("CloudFoundy Applications  {{.Used}}/{{.Limit}} used",
			map[string]interface{}{
				"Used":  formattedGB(orgUsage.TotalMemoryUsed()),
				"Limit": formattedGB(cmd.cf.CurrentOrganization().QuotaDefinition.InstanceMemoryLimitInMB)}), color.New(color.FgYellow, color.Bold)))

```
- When a `messageID` does not exist in the translation files for a given language other than the default (English). The default message will be used
- When a `messageID` does not exist in any of the translation file. A panic error will be thrown with the error message format: `"<MESSAGE-ID>" not found in language "<language>"`. This panic behavior is provided by go-i18n. If we want to just return an error and not panic I can adjust this accordingly to handle the panic as an error only if the error message is of this type.

# Steps to Test
1. Open a terminal and change directory to the root of the `ibm-cloud-cli-sdk` local repo
2. Run the command `go test ./...`
3. Verify all tests pass

Aerex commented on bin/generate-language-resources:
```diff
@@ -3,8 +3,9 @@
+pushd resources 1>/dev/null
   go fmt ./...
-popd
+popd 1>/dev/null
```
**Callout:** Added `1>/dev/null` to suppress printing out popped directory

Aerex commented on i18n/i18n.go:
```diff
@@ -1,32 +1,97 @@
 
-var T i18n.TranslateFunc
+var (
+	bundle *i18n.Bundle
```
**Callout**: Added `bundle` variable that will handle initializing the translate function and setting the default langugae.

Aerex commented on i18n/i18n.go:
```diff
@@ -1,32 +1,97 @@
-	defaultTfunc := i18n.MustTfunc(defaultLocale)
-	supportedLocales := supportedLocales()
+// Bundle returns an instane of i18n.bundle
+func Bundle() *i18n.Bundle {
+	if bundle == nil {
+		bundle = i18n.NewBundle(language.English)
+		bundle.RegisterUnmarshalFunc("json", json.Unmarshal)
+	}
+	return bundle
+}
```
**Callout**: Created `Bundle()` method for easy of retrieval of the singleton

Aerex commented:
Testing new changes with CLI and Dev plugin. Will add a do-no-merge label until done. FYI @steveclay

steveclay commented on docs/plugin_developer_guide.md:
```diff
@@ -1111,6 +1111,8 @@ You can find other examples in [tests](https://github.com/maxbrunsfeld/counterfe
 
 IBM Cloud CLI tends to be used globally. Both IBM Cloud CLI and its plug-ins should support globalization. We have enabled internationalization (i18n) for CLI's base commands with the help of the third-party tool "[go-i18n](https://github.com/nicksnyder/go-i18n)". To keep user experience consistent, we recommend plug-in developers follow the CLI's way of i18n enablement.
 
+Please install the *go-i18n* CLI for version 1.10.0. Newer versions of the CLI are no longer compatible with translations files prior to go-i18n@2.0.0. You can install the CLI using the command: `go install github.com/nicksnyder/go-i18n/goi18n@1.10.1`
```
I'm not clear on why we are having users install the old version of go-i18n. Is this always necessary?

Aerex replied:
Yes. The new version of the go-i18n generates translation files using a different json structure. I checked both versions again and got this result.

**Generated JSON from en_US in cli-dv-plugin using go-i18n CLI**

*Using V2*
```
{
  "Could not create the service instance.": {
    "hash": "sha1-xxxx",
      "other": "Could not create the service instance"
  }
  ...
}
```

*Using V1*
```
{
  "id": "Could not create the service instance",
  "translation": "Could not create the service instance"
}
```

steveclay commented on i18n/i18n.go:
```diff
@@ -70,21 +132,39 @@ func loadAsset(assetName string) {
 
-	err = i18n.ParseTranslationFileBytes(assetName, bytes)
-	if err != nil {
+	if _, err := bundle.ParseMessageFileBytes(bytes, assetName); err != nil {
```
Looks to be defining a second "err" variable in the same context. This would give inconsistent results.

Aerex replied:
Can you elaborate? This method can return two different errors and I think we should panic with the correct error.

steveclay replied:
Line 135 is creating a variable named `err`, but this variable was already created on line 130. Golang will let you create 2 variables with the same name in a scope, but the results you get from this can be unpredictable.

Aerex replied:
Oh I see.  Yeah let me fix that

steveclay commented on i18n/i18n.go:
```diff
@@ -1,32 +1,105 @@
+		// the translation could not be found in any of the files
+		// panic and abort
+		if msg == "" {
+			panic(err)
```
Would we have to panic here, or could we return something like "Message not found" ? I think the design of i18n is to panic, but might we deviate here to prevent a potentially ugly failure for customers?

steveclay replied:
Yeah, good point

steveclay reviewed (changes requested):
One change and one question to resolve before we can complete this.

Aerex commented:
@steveclay I have updated the code to use **!!I18N_MESSAGE_NOT_FOUND!!** whenever the translation can not be found in any of the translation files. Also, updated the unit tests accordingly
